oauth-common/src/main/java/io/strimzi/kafka/oauth/common/HttpUtil.java

```diff
@@ -0,0 +1,121 @@
+        return postOrGet(uri, socketFactory, null, authorization, null, null, responseType);
+    }
+
+    public static <T> T get(URI uri, SSLSocketFactory socketFactory, HostnameVerifier hostnameVerifier, String authorization, Class<T> responseType) throws IOException {
+        return postOrGet(uri, socketFactory, hostnameVerifier, authorization, null, null, responseType);
+    }
+
+    public static <T> T post(URI uri, String authorization, String contentType, String body, Class<T> responseType) throws IOException {
+        return postOrGet(uri, null, null, authorization, contentType, body, responseType);
+    }
+
+    public static <T> T post(URI uri, SSLSocketFactory socketFactory, String authorization, String contentType, String body, Class<T> responseType) throws IOException {
+        return postOrGet(uri, socketFactory, null, authorization, contentType, body, responseType);
+    }
+
+    public static <T> T post(URI uri, SSLSocketFactory socketFactory, HostnameVerifier verifier, String authorization, String contentType, String body, Class<T> responseType) throws IOException {
+        return postOrGet(uri, socketFactory, verifier, authorization, contentType, body, responseType);
+    }
+
+    public static <T> T postOrGet(URI uri, SSLSocketFactory socketFactory, HostnameVerifier hostnameVerifier, String authorization, String contentType, String body, Class<T> responseType) throws IOException {
+        HttpURLConnection con;
+        try {
+            con = (HttpURLConnection) uri.toURL().openConnection();
+        } catch (MalformedURLException e) {
+            throw new IllegalArgumentException("Malformed token endpoint url: " + uri);
+        }
+
+        if (con instanceof HttpsURLConnection) {
+            HttpsURLConnection scon = (HttpsURLConnection) con;
+            if (socketFactory != null) {
+                scon.setSSLSocketFactory(socketFactory);
+            }
+            if (hostnameVerifier != null) {
+                scon.setHostnameVerifier(hostnameVerifier);
+            }
+        } else if (socketFactory != null) {
+            log.warn("SSL socket factory set but url scheme not https ({})", uri);
+        }
+
+        con.setUseCaches(false);
+        if (body != null) {
+            con.setDoOutput(true);
+        }
+
+        con.setRequestMethod(body != null ? "POST" : "GET");
+        if (authorization != null) {
+            con.setRequestProperty("Authorization", authorization);
+        }
+        con.setRequestProperty("Accept", "application/json");
+        if (body != null && body.length() > 0) {
+            if (contentType == null) {
+                throw new IllegalArgumentException("contentType must be set when body is not null");
+            }
+            con.setRequestProperty("Content-Type", contentType);
+        }
+
+        try {
+            con.connect();
```

Comment:
I assume you're not calling disconnect because it's reasonable to assume that if OAuth is configured we'll be regularly connecting to the authentication server. That's probably worth a comment.

Reply:
AFAIK java.net.HttpURLConnection caches connections behind the scenes. connect() call actually takes available connection from internal pool if one exists. And calling close would simply free it and return it to the pool where after some idle time it would be closed.

Also of note - OAuth2 token endpoint is used only during Kafka session initiation, not during production / consumption of records.

Reply:
```
    /**
     * Indicates that other requests to the server
     * are unlikely in the near future. Calling disconnect()
     * should not imply that this HttpURLConnection
     * instance can be reused for other requests.
     */
    public abstract void disconnect();
```

My point is either you're not calling disconnect on purpose (because you're expecting to benefit from the persistent connection to the server while it remains open/in the pool), in which case you should explain why in a comment, or you're not calling disconnect by accident in which case you should fix it.

Reply:
Added comments in code and JavaDoc.